src/PrestaShopBundle/Entity/Repository/AdminFilterRepository.php

```diff
@@ -0,0 +1,113 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Entity\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use PrestaShopBundle\Entity\AdminFilter;
+
+class AdminFilterRepository extends EntityRepository
+{
+    /**
+     * @param $employeeId
+     * @param $shopId
+     * @param $controller
+     * @param $action
+     *
+     * @return AdminFilter|null
+     */
+    public function findByEmployeeAndRouteParams($employeeId, $shopId, $controller, $action)
+    {
+        return $this->findOneBy([
+            'employee' => $employeeId ?: 0,
+            'shop' => $shopId ?: 0,
+            'controller' => $controller,
+            'action' => $action,
+        ]);
+    }
+
+    /**
+     * @param $employeeId
+     * @param $shopId
+     * @param $controller
+     * @param $action
+     *
+     * @throws \Doctrine\ORM\ORMInvalidArgumentException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     *
+     * @return bool Returns false if entity was not found
+     */
+    public function removeByEmployeeAndRouteParams($employeeId, $shopId, $controller, $action)
+    {
+        $adminFilter = $this->findOneBy([
+            'employee' => $employeeId ?: 0,
+            'shop' => $shopId ?: 0,
+            'controller' => $controller,
+            'action' => $action,
+        ]);
+
+        if (null !== $adminFilter) {
+            $this->getEntityManager()->remove($adminFilter);
+            $this->getEntityManager()->flush();
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $employeeId
+     * @param $shopId
+     * @param $filters
+     * @param $controller
+     * @param $action
+     * @throws \Doctrine\ORM\ORMInvalidArgumentException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    public function createOrUpdateByEmployeeAndRouteParams($employeeId, $shopId, $filters, $controller, $action)
+    {
+        $adminFilter = $this->findOneBy([
+            'employee' => $employeeId,
+            'shop' => $shopId,
+            'controller' => $controller,
+            'action' => $action,
+        ]);
+
+        $adminFilter = is_null($adminFilter) ? new AdminFilter() : $adminFilter;
+
+        $adminFilter->setAction($action)
+            ->setController($controller)
+            ->setEmployee($employeeId)
+            ->setShop($shopId)
+            ->setFilter(json_encode($filters))
```

Comment:
`setFilter` may json_encode itself, and `getFilter` json_decode. Could be better if someone manage filters in another way

Reply:
As theses functions are also used in Product pages, I don't want to risk regressions right now.
Maybe a "refacto" pr to improve the system is worth, but I'd like to keep this one scoped :)